pkg/utils/utils.go

```diff
@@ -169,3 +175,62 @@ func AppendPortIfNotExists(slice []v1.ContainerPort, elems ...v1.ContainerPort)
 	}
 	return slice
 }
+
+// IsCrdAvailable checks if a given CRD is present in the cluster by verifying the
+// existence of its API.
+func IsCrdAvailable(config *rest.Config, groupVersion, kind string) (bool, error) {
+	gvResources, err := GetAvailableResourcesForApi(config, groupVersion)
+	if err != nil {
+		return false, err
+	}
+
+	found := false
+	if gvResources != nil {
+		for _, crd := range gvResources.APIResources {
+			if crd.Kind == kind {
+				found = true
+				break
+			}
+		}
+	}
+
+	return found, nil
+}
+
+// GetAvailableResourcesForApi returns the list of discovered resources that belong
+// to the API specified in groupVersion. The first query to a specifig groupVersion will
+// query the cluster API server to discover the available resources and the discovered
+// resources will be cached and returned to subsequent invocations to prevent additional
+// queries to the API server.
+func GetAvailableResourcesForApi(config *rest.Config, groupVersion string) (*metav1.APIResourceList, error) {
+	var gvResources *metav1.APIResourceList
+	var ok bool
+
+	if gvResources, ok = gvResourcesCache[groupVersion]; !ok {
+		discoveryClient, newClientErr := discovery.NewDiscoveryClientForConfig(config)
+		if newClientErr != nil {
+			return nil, newClientErr
+		}
+
+		var getGvResourcesErr error
+		gvResources, getGvResourcesErr = discoveryClient.ServerResourcesForGroupVersion(groupVersion)
+		if getGvResourcesErr != nil && !apierr.IsNotFound(getGvResourcesErr) {
+			return nil, getGvResourcesErr
+		}
+
+		SetAvailableResourcesForApi(groupVersion, gvResources)
```

Comment:
Is SetAvailableResourcesForApi used outside this package?
If not you could, instead create a func to store it, just put them in the cache here.

Reply:
ok, it is. :)